src/postgres_client.rs

```diff
@@ -1237,13 +1262,29 @@ pub struct PostgresClientBuilder {}
 impl PostgresClientBuilder {
     pub fn build_pararallel_postgres_client(
         config: &GeyserPluginPostgresConfig,
-    ) -> Result<ParallelPostgresClient, GeyserPluginError> {
-        ParallelPostgresClient::new(config)
-    }
+    ) -> Result<(ParallelPostgresClient, Option<u64>), GeyserPluginError> {
+        let batch_optimize_by_skiping_older_slots = match config.batch_optimize_by_skiping_old_slots
+        {
+            true => {
+                //one request client
+                let mut on_load_client = SimplePostgresClient::new(config)?;
+
+                let max_queue_size = u64::try_from(MAX_ASYNC_REQUESTS).unwrap();
```

Comment:
I don't understand how this number is picked? Why MAX_ASYNC_REQUESTS?

Reply:
I do not know details how geyser plugin is called by solana-validator. So this is why I created some "safe scope".

If the order in  [`receiver`](https://github.com/solana-labs/solana-accountsdb-plugin-postgres/blob/b9418aff990ecb32b4031a774bc8daac8a8a9a42/src/postgres_client.rs#L1037) queue is sorted by slot and guarantee by solana-validator then this  `threads` (from config file). Because this is max parallel. I just here chosen "safer value" - and count that you will help with choose the correct one. 

Also I think that when validator is loading snapshoot it starts populate it from newest to oldest. This will not corrupt data correctness (in DB) in case of restart during loading just because `slots_at_startup` HashSet are write at the end of this process. But flush `slots_at_startup` is implementation details and could be easily changed by further PR and this will silently make this code incorrect in such situation. Maybe it would be good to set also is_startup flag in DB. This will also indicate the readers of DB that they can expect delay of data.

Reply:
I understand the cushion reasoning. Just do not like using MAX_ASYNC_REQUESTS as to os rather unrelated. Can you define a new variable like SAFE_BATCH_STARTING_SLOT_CUSHION?

Reply:
Ok. But what value give to that variable?

I think that  `SAFE_BATCH_STARTING_SLOT_CUSHION` could be just  the number of workers (`thread` value from config). But to make it correct this assertions must be met:
- slots in writing queue must be ordered.
- the queue contains also accounts, transaction. It must be guarantee that queue order is like this: for slot x, we insert first slot x then all accounts and transactions for slot x.

If this is not guarantee  so I am not sure how to choose **!!safe**  `SAFE_BATCH_STARTING_SLOT_CUSHION` value.

Reply:
I have created variable as suggested. And put there some big number that should be enough. Let me know if tis is ok now